src/extract.rs

```diff
@@ -168,3 +168,38 @@ impl ToTokens for InstArguments {
         self.0.to_tokens(tokens)
     }
 }
+
+pub struct TestAttrs(pub(crate) Punctuated<NestedMeta, Token![,]>);
+
+impl TestAttrs {
+    pub fn try_extract(item: &mut ItemMod) -> syn::Result<Option<Self>> {
+        for (pos, attr) in item.attrs.iter().enumerate() {
+            if attr.path.is_ident("test_attributes") {
+                match attr.style {
+                    AttrStyle::Outer => {}
+                    AttrStyle::Inner(_) => {
+                        return Err(Error::new_spanned(attr, "cannot be an inner attribute"))
+                    }
+                };
+                let args = match attr.parse_meta()? {
```

Comment:
I'm not a big fan of using `syn`'s `Meta` types, since they're based on the old limited attribute parsing that rust used to have (_e.g._, incapable of handling `doc = include_str!("...")`, or, in the case of `generic-tests`, maybe, `should_panic(expected = some_macro!())`), but since in the case of test attributes, the macro case is very contrived, we can get away with meta.

  - So this is mostlya word of warning for _future proc-macro code_ (for instance, I'm more of a fan of using [`.parse_args()`](https://docs.rs/syn/1.0.98/syn/struct.Attribute.html#method.parse_args) with custom `Parse` types if necessary, or in this case, a `TokenStream` to act as a transparent passthrough could have worked too)

Reply:
Oh, I wasn't aware of this distinction! Thanks for explaining.

I mean, I'd be happy with a transparent `TokenStream` passthrough in this case, and I doubt it would be a huge deal to implement that, so I can give it a go!